pandas/_libs/lib.pyx

```diff
@@ -134,8 +135,8 @@ def is_scalar(val: object) -> bint:
             or util.is_period_object(val)
             or is_decimal(val)
             or is_interval(val)
-            or util.is_offset_object(val))
-
+            or util.is_offset_object(val)
+            or np.isscalar(val))
```

Comment:
Does numpy's C-API have an implementation of this?

Reply:
Does the behavior vary across supported versions of numpy?

Reply:
> Does numpy's C-API have an implementation of this?

I went [this way](https://github.com/numpy/numpy/blob/b7ff1d3a54e2e69c82df24b658e545606f4db8e7/numpy/core/numeric.py#L1931), @jbrockmendel 
Further, `ScalarType` is in numerictypes.py [here](https://github.com/numpy/numpy/blob/b7ff1d3a54e2e69c82df24b658e545606f4db8e7/numpy/core/numerictypes.py#L515).

Reply:
I would just import Number / Fraction here and do an isinstance check